### PR TITLE
Added Reissner-Nordstorm metric in Vaccum_solutions.py

### LINKED ISSUE #309: Add loads of predefined metrics in symbolic module
🐞 **What bugs you?**
  - Have predefined metrics in *symbolic* module

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by obi-wan-shinobi**
Added the Reissner-Nordstorm metric
Addresses #309 

@ritzvik 
@shreyasbapat

### CODE REVIEW

**Comment by ritzvik** [comment]
move this to `vacuum_solutions.py` :)

**Comment by ritzvik** [comment]
Kindly Add tests in `tests/test_symbolic/test_predefined`

**Comment by obi-wan-shinobi** [comment]
Okay!

File: src/einsteinpy/symbolic/predefined/vacuum_solutions.py [hunk 1]
```diff
@@ -115,3 +115,44 @@ def KerrNewman(
     ).tolist()
     metric[0][3] = metric[3][0] = sch * r * a * (sin(theta) ** 2) / (Sigma * c)
     return MetricTensor(metric, coords, "ll", name="KerrNewmanMetric")
+
+
+def Reissner_Nordstorm(
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/predefined/vacuum_solutions.py]:
```suggestion
def ReissnerNordstorm(
```
drop the underscore, as followed by other predefined metrics

File: src/einsteinpy/symbolic/predefined/vacuum_solutions.py [hunk 2]
```diff
@@ -115,3 +115,44 @@ def KerrNewman(
...
+    """
+    coords = symbols("t r theta phi")
+    t, r, theta, phi = coords
+    rQsq = ((Q ** 2) * G) / (4 * pi * eps_0 * (c ** 4))
+    Arn = 1 - sch / r + rQsq / r ** 2
+
+    metric = diag(
+        (Arn * (c ** 2)), -(1 / Arn), -(r ** 2), -(r ** 2) * sin(theta) ** 2,
+    ).tolist()
+    return MetricTensor(metric, coords, "ll", name="Reissner_NordstormMetric")
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/predefined/vacuum_solutions.py]:
```suggestion
    return MetricTensor(metric, coords, "ll", name="ReissnerNordstormMetric")
```

File: src/einsteinpy/symbolic/predefined/vacuum_solutions.py [hunk 3]
```diff
@@ -115,3 +115,44 @@ def KerrNewman(
...
+        Defaults to ``Q``.
+
+    """
+    coords = symbols("t r theta phi")
+    t, r, theta, phi = coords
+    rQsq = ((Q ** 2) * G) / (4 * pi * eps_0 * (c ** 4))
+    Arn = 1 - sch / r + rQsq / r ** 2
+
+    metric = diag(
+        (Arn * (c ** 2)), -(1 / Arn), -(r ** 2), -(r ** 2) * sin(theta) ** 2,
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/predefined/vacuum_solutions.py]: I guess you would have to divide all the elements by `(c ** 2)`. This is maintain homogeneity with other metrics in `vacuum_solutions.py`. 
Reissner–Nordström is actually a special(limited case `a=0`) of Kerr-Newman metric which is defined above. Make sure that both are compatible.

Comment (ritzvik) [reply src/einsteinpy/symbolic/predefined/vacuum_solutions.py]: Add the new metric to `__init__.py` also

Comment (obi-wan-shinobi) [reply src/einsteinpy/symbolic/predefined/vacuum_solutions.py]: ok

**Comment by ritzvik** [comment]
For tests, add the entry here
https://github.com/einsteinpy/einsteinpy/blob/2f8b5b38e97fbdfb798632e046b5d81bf19a3015/src/einsteinpy/tests/test_symbolic/test_predefined/test_all.py#L27-L51

Also, because Reissner–Nordström is a special case of Kerr-Newman, one addition is also needed here.

https://github.com/einsteinpy/einsteinpy/blob/2f8b5b38e97fbdfb798632e046b5d81bf19a3015/src/einsteinpy/tests/test_symbolic/test_predefined/test_all.py#L54-L63

**Comment by ritzvik** [comment]
Don't forget to lint the code

Squash n rebase after this https://github.com/einsteinpy/einsteinpy/wiki/How-to-refactor-commits

**Comment by ritzvik** [comment]
The problem may be in definition of the metric

**Comment by obi-wan-shinobi** [comment]
> The problem may be in definition of the metric

Yep, working on it :)

**Comment by ritzvik** [comment]
:( I'll see to it, in a while

**Comment by obi-wan-shinobi** [comment]
> :( I'll see to it, in a while

Thank you! I think I complicated things. I'll try the derivaition before making further changes. :0

**Comment by ritzvik** [comment]
Okay, this seems working

**Comment by ritzvik** [comment]
I'll give it one more look tomorrow before merging. Squash n rebase meanwhile

https://github.com/einsteinpy/einsteinpy/wiki/How-to-refactor-commits

**Pull request closed without merging**